nixos/modules/system/boot/systemd.nix

```diff
@@ -827,6 +827,27 @@ in
 
     system.build.units = cfg.units;
 
+    # Systemd provides various NSS modules to look up dynamic users, locally
+    # configured IP adresses and local container hostnames.
+    # On NixOS, these can only be passed to the NSS system via nscd (and its
+    # LD_LIBRARY_PATH), which is why it's usually a very good idea to have nscd
+    # enabled (also see the config.nscd.enable description).
+    # While there is already an assertion in place complaining loudly about
+    # having nssModules configured and nscd disabled, for some reason we still
+    # check for nscd being enabled before adding to nssModules.
```

Comment:
@florianjacob as the author of e370e97f3dab410ac460af3cb9974b8df7b3214d, can you elaborate on why we silently disable these nss modules if nscd is enabled, even though there's an assertion in https://github.com/NixOS/nixpkgs/pull/86940/files#diff-5796c52b71eee35842f408f4126430d6R126-R127 that should complain if nscd is enabled?

Maybe instead of silently ignoring these (and breaking dynamic user support, as well as other NSS modules), can't we ask the user to `mkForce system.nssModules = []` in the assertion message if they really doesn't want any external NSS modules?

Reply:
Also see https://github.com/NixOS/nixpkgs/issues/43607 and https://github.com/NixOS/nixpkgs/pull/86010 for context.

Reply:
I really like your idea. There is no reason for the current behaviour and `system.nssModules = systemd.out` seems like the most elegant solution and also prevents users from accidentially breaking their systems by disabling nscd.

Reply:
I hope @florianjacob can shed some light on this. I think we should never have to add vague comments like `for some reason we still check for nscd being enabled before adding to nssModules.` The source should be a place of truth :) If we can't figure it out and the original authors don't respond we might as well change the implementation until we encounter errors and can properly document them.